frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
I believe calculating the checksum at the very first time a book has been opened is important. No matter whether sync feature has been enabled or not. Though I can still move this line into KOSync:init. i.e. the checksum will not be calculated if kosync plugin has been disabled or removed. Users cannot quite understand the relationship between document saving feature and progress sync feature, and if one user read a book on a device without caring the document saving feature. Then sometime later, he decided to enable progress sync feature to sync to another device. He would find the progress could not be synced for a same file (at least same file name). This can be happened pretty randomly, such as adding a highlight on one device. That's the reason we should calculate the checksum at the very first time a document has been opened.